### PR TITLE
Commit of Blog feature code

### PR DESCRIPTION
**Pull request by fabriceumuhire**
### What the PR does
- CRUD articles
### Tasks description
- Create article using POST method and store the image to cloudinary 
- Retrieve articles or one article by id from the DB
- Update article and store in the DB
- Delete article from DB and image from Cloudinary
### How to test
- Use POSTMAN or MongoDB Compass to view all queries
- Check URL from the article if they display images.

### CODE REVIEW

File: server/src/routes/articles.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+router.delete("/articles/:id/:publicId", async (req, res) => {
+    try {
+      cloudinary.v2.uploader.destroy(req.params.publicId, async (error,result) => {
+        console.log(error);
+        await Blog.findByIdAndRemove({
+          _id: req.params.id,
+          publicId: result.public_id,
+        });
+        res.status(200).send();
```
Comment (niyodusengaclement) [inline server/src/routes/articles.js]: Return the message to the user that the article deleted successfully

File: server/src/routes/articles.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    
+  } catch {
+    res.status(404);
+    res.send({ error: "Article doesn't exist!" });
+  }
+});
+
+router.delete("/articles/:id/:publicId", async (req, res) => {
+    try {
+      cloudinary.v2.uploader.destroy(req.params.publicId, async (error,result) => {
```
Comment (niyodusengaclement) [inline server/src/routes/articles.js]: You should check if the article id found then u delete if not u return the message that the article not found

**Review by niyodusengaclement — CHANGES_REQUESTED** [review]
Resolve these issues

File: server/src/routes/articles.js (hunk 2 above)
Comment (fabriceumuhire) [reply server/src/routes/articles.js]: Added in the catch in case article id is not  found @niyodusengaclement

Comment (fabriceumuhire) [reply server/src/routes/articles.js]: @niyodusengaclement I have added authentication and validation as well message to be displayed. Kindly review and let me know if I can move forward with testing.

File: server/src/controllers/articles.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    } catch {
+        res.status(404);
+        res.send({ error: "Article doesn't exist!" });
+    }
+
+};
+
+exports.deleteOne = async (req, res) => {
+    try {
+        cloudinary.v2.uploader.destroy(req.params.publicId, async (error,result) => {
```
Comment (niyodusengaclement) [inline server/src/controllers/articles.js]: 1st check if the id exists in DB, if exists return its information including publicId withoud passing public id in params. Just provide id and check if exists if not return not found msg

File: server/src/middleware/verify.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+const jwtoken = require("jsonwebtoken");
+
+module.exports = function (req, res, next) {
+    const token = req.header("auth-token");
+    if(!token) {
+        res.status(401).send("Not authorization");
+    }
+    try {
+        const verification = jwtoken.verify(token, "jsjhdqsdjlqhq");
```
Comment (niyodusengaclement) [inline server/src/middleware/verify.js]: use .env for store JWT screte key or other confidential values

**Review by niyodusengaclement — CHANGES_REQUESTED** [review]
I've mentioned some of the many bugs. Fix em all and let me know

File: server/src/controllers/articles.js (hunk 3 above)
Comment (fabriceumuhire) [reply server/src/controllers/articles.js]: @niyodusengaclement Done for both checking if id exist as well as passing the message if id doesn't exist with corresponding message.

**Pull request merged**